### PR TITLE
fix: loading state error type to be compatible with redis

### PR DESCRIPTION
**Pull request by kostasrim**
addresses #2617 

* add -LOADING prefix for loading errors
* replace -ERR with -LOADING for loading errors

### CODE REVIEW

File: src/facade/reply_builder.cc [hunk 1]
```diff
@@ -296,17 +297,26 @@ void RedisReplyBuilder::SendError(string_view str, string_view err_type) {
...
+  }
+
+  if (err_type == kLoadingErrType) {
+    iovec v[] = {IoVec(kLoadingPrefix), IoVec(str), IoVec(kCRLF)};
     Send(v, ABSL_ARRAYSIZE(v));
+    return;
   }
+
+  iovec v[] = {IoVec(kErrPref), IoVec(str), IoVec(kCRLF)};
+  Send(v, ABSL_ARRAYSIZE(v));
```
Comment (dranikpg) [inline src/facade/reply_builder.cc:319]: The inconsistency is that we don't use error types for prefixes for other cases (I don't even remember why we have them) and use the prefix directly, see `kAuthRejected` or `kScriptNotFound`... so what would be the preferred way now 🙂

Comment (kostasrim) [reply src/facade/reply_builder.cc]: yes I see. error types are used for stats, you might want to send a different error message but register it under the same stat name

File: src/server/main_service.cc [hunk 2]
```diff
@@ -988,7 +988,8 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
 
   // Check if the command is allowed to execute under this global state
   bool allowed_by_state = true;
-  switch (etl.gstate()) {
+  const auto gstate = etl.gstate();
```
Comment (romange) [inline src/server/main_service.cc:991]: nit: can you please replace auto with GlobalState?
auto is great for generic code, avoid 120 char type names etc. But lets not use it at least for POD types - it's much harder to review the code when you do not know what type the variable is.

Comment (kostasrim) [reply src/server/main_service.cc]: done :)

**Review by dranikpg — APPROVED** [review]

**Pull request merged**